pvnet_app/app.py

```diff
@@ -314,39 +197,39 @@ def app(
     # Prepare all the models which can be run
     forecast_compilers = {}
     data_config_paths = []
-    for model_key, model_config in model_to_run_dict.items():
+    for model_config in model_configs:
         # First load the data config
         data_config_path = PVNetBaseModel.get_data_config(
-            model_config["pvnet"]["name"],
-            revision=model_config["pvnet"]["version"],
+            model_config.pvnet.repo,
+            revision=model_config.pvnet.version,
         )
 
         # Check if the data available will allow the model to run
         model_can_run = check_model_satellite_inputs_available(data_config_path, t0, sat_datetimes)
 
         if model_can_run:
             # Set up a forecast compiler for the model
-            forecast_compilers[model_key] = ForecastCompiler(
-                model_tag=model_key,
-                model_name=model_config["pvnet"]["name"],
-                model_version=model_config["pvnet"]["version"],
-                summation_name=model_config["summation"]["name"],
-                summation_version=model_config["summation"]["version"],
+            forecast_compilers[model_config.name] = ForecastCompiler(
+                model_tag=model_config.name,
```

Comment:
It might be cleaner here just to pass in the model config rather than unpack all of its attributes but I'm not sure. What do you think?

Reply:
oh, thats a great idea. Thats advantage of making a nice configuration pydantic model